src/goto-instrument/wmm/event_graph.h

```diff
@@ -51,40 +53,72 @@ class event_grapht
     std::string print_name(const critical_cyclet &redyced,
       memory_modelt model) const;
 
-    bool check_AC(const_iterator s_it, const abstract_eventt &first,
+    bool check_AC(
+      data_typet::const_iterator s_it,
+      const abstract_eventt &first,
       const abstract_eventt &second) const;
-    bool check_BC(const_iterator it, const abstract_eventt &first,
+    bool check_BC(
+      data_typet::const_iterator it,
+      const abstract_eventt &first,
       const abstract_eventt &second) const;
 
   public:
     unsigned id;
-
     bool has_user_defined_fence;
 
+    using iterator=data_typet::iterator;
+    using const_iterator=data_typet::const_iterator;
+    using value_type=data_typet::value_type;
+
+    iterator begin() { return data_.begin(); }
+    const_iterator begin() const { return data_.begin(); }
+    const_iterator cbegin() const { return data_.cbegin(); }
+
+    iterator end() { return data_.end(); }
+    const_iterator end() const { return data_.end(); }
+    const_iterator cend() const { return data_.cend(); }
+
+    template <typename T>
+    void push_front(T &&t) { data_.push_front(std::forward<T>(t)); }
+
+    template <typename T>
+    void push_back(T &&t) { data_.push_back(std::forward<T>(t)); }
+
+    value_type &front() { return data_.front(); }
+    const value_type &front() const { return data_.front(); }
+
+    value_type &back() { return data_.back(); }
+    const value_type &back() const { return data_.back(); }
+
+    size_t size() const { return data_.size(); }
+
     critical_cyclet(event_grapht &_egraph, unsigned _id)
       :egraph(_egraph), id(_id), has_user_defined_fence(false)
     {
     }
 
     void operator()(const critical_cyclet &cyc)
     {
-      clear();
-      for(const_iterator it=cyc.begin(); it!=cyc.end(); it++)
-        push_back(*it);
+      data_.clear();
+      for(
+        auto it=cyc.data_.cbegin();
+        it!=cyc.data_.cend();
+        ++it)
+        data_.push_back(*it);
```

Comment:
Isn't there a more concise way of expressing this instead of an explicit `for` loop? (Criticism of the original code, not the change.)